azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -905,19 +911,15 @@ public void resume(String user) {
   }
 
   public void kill(String user) {
-    synchronized (mainSyncObj) {
-      logger.info("Flow killed by " + user);
-      flow.setStatus(Status.KILLED);
-      kill();
-      updateFlow();
-    }
-    interrupt();
+    logger.info("Flow killed by " + user);
+    kill();
   }
 
-  private void kill() {
+  public void kill() {
     synchronized (mainSyncObj) {
```

Comment:
`mainSyncObj` is not even a final member here. So, locking on it seems unsafe and not a best practice.